beauty-consulting-app/app/assets/components/EditProfileViewModal.js

```diff
@@ -0,0 +1,199 @@
+import React, { useState, useEffect, useContext } from "react";
+import {
+	Modal,
+	View,
+	Text,
+	TouchableOpacity,
+	StyleSheet,
+	TextInput,
+} from "react-native";
+import globalStyles from "../GlobalStyles";
+import AgeIcon from "../images/birthday-cake.svg";
+import PhoneIcon from "../images/phone-call.svg";
+import EmailIcon from "../images/email.svg";
+import GenderIcon from "../images/genders.svg";
+import DateTimePicker from "@react-native-community/datetimepicker";
+import { UserContext, updateUserContext } from "../../contexts/userContext";
+import handleHTTPError from "utils/errorHandling";
+import api from "utils/axios";
+
+const EditProfileViewModal = ({ visible, onClose, profileDetails }) => {
+	const [formData, setFormData] = useState({
+		birthday: new Date(), // default to current date
+		gender: "",
+		phoneNumber: "",
+		email: "",
+	});
+
+	const userContext = useContext(UserContext);
+
+	useEffect(() => {
+		if (profileDetails) {
+			setFormData({
+				birthday: new Date(profileDetails.birthday),
+				gender: profileDetails.gender,
+				phoneNumber: profileDetails.phoneNumber,
+				email: profileDetails.email,
+			});
+		}
+	}, [profileDetails]);
+
+	const handleInputChange = (field, value) => {
+		setFormData((prev) => ({
+			...prev,
+			[field]: value,
+		}));
+	};
+
+	const handleDateChange = (event, selectedDate) => {
+		if (event.type === "set") {
+			const currentDate = selectedDate || formData.birthday;
+			setFormData((prev) => ({
+				...prev,
+				birthday: currentDate,
+			}));
+		}
+	};
+
+	const saveAndClose = async () => {
+		updatedData = {
+			...userContext,
+			email: formData.email,
+			info: {
+				phoneNumber: formData.phoneNumber,
+				gender: formData.gender,
+				birthday: formData.birthday,
+			},
+		};
+		userContext.updateUserContext(updatedData);
+		try {
+			await api.put(`client/${userContext.username}`, updatedData);
```

Comment:
Ideally, we would have some input validation here with Joi like we do in the client signup process. Just noting this down for you to see. I think we can add validation in a separate self-contained PR later.